### ISSUE TITLE
Posts containing embed links to other posts fail to download

### ISSUE THREAD

**Issue by asdf1324** [xuejianxianzun/PixivFanboxDownloader#32]
Example post (free): /@singlecask/posts/4012918

![image](https://user-images.githubusercontent.com/19871820/175548051-415ad9e2-a7a9-4bf4-9bcf-06310cd0f07e.png)

![image](https://user-images.githubusercontent.com/19871820/175547860-03e7e55b-ef22-401f-bd03-ce430ddf5af6.png)

With the downloader configured to save links in the post body / the actual text of the body, posts containing embeds fail to download with an `TypeError: Cannot read properties of undefined (reading 'match')` error here:

![image](https://user-images.githubusercontent.com/19871820/175558711-2878f2b2-02a7-4959-893d-1ee3d0643c15.png)

![image](https://user-images.githubusercontent.com/19871820/175558808-0581465c-6f1c-46be-98d7-81e62a6bc65f.png)

Quick workaround to get it going, temporarily:

![image](https://user-images.githubusercontent.com/19871820/175558154-a465651a-80ff-4a67-a9e3-78512ab68368.png)

**Comment by xuejianxianzun** [comment]
Please find the data of this post in the network and copy it here

**Comment by xuejianxianzun** [comment]
E.g
![QQ图片20220624224553.png](https://s2.loli.net/2022/06/24/iaVvgezLbsG2XSP.png)

**Comment by asdf1324** [comment]
FWIW, here's the response from GET /post.info?postId=4012918

```json
{
    "body": {
        "id": "4012918",
        "title": "【全体公開】バニーガール円香",
        "coverImageUrl": "https://pixiv.pximg.net/c/1200x630_90_a2_g5/fanbox/public/images/post/4012918/cover/u81tmvUMpM0Za5Jk6pkyjSM0.jpeg",
        "feeRequired": 0,
        "publishedDatetime": "2022-06-24T19:00:00+09:00",
        "updatedDatetime": "2022-06-24T19:00:00+09:00",
        "type": "article",
        "body": {
            "blocks": [
                {
                    "type": "header",
                    "text": "ちょっとだけ高画質です"
                },
                {
                    "type": "image",
                    "imageId": "CkHy34pgij110KHnC8dE24v5"
                },
                {
                    "type": "p",
                    "text": "有料プランはこちら↓"
                },
                {
                    "type": "url_embed",
                    "urlEmbedId": "SwQufnahqlm2w1HQPBO4Gop5"
                },
                {
                    "type": "p",
                    "text": ""
                }
            ],
            "imageMap": {
                "CkHy34pgij110KHnC8dE24v5": {
                    "id": "CkHy34pgij110KHnC8dE24v5",
                    "extension": "jpeg",
                    "width": 1066,
                    "height": 1974,
                    "originalUrl": "https://downloads.fanbox.cc/images/post/4012918/CkHy34pgij110KHnC8dE24v5.jpeg",
                    "thumbnailUrl": "https://downloads.fanbox.cc/images/post/4012918/w/1200/CkHy34pgij110KHnC8dE24v5.jpeg"
                }
            },
            "fileMap": {},
            "embedMap": {},
            "urlEmbedMap": {
                "SwQufnahqlm2w1HQPBO4Gop5": {
                    "id": "SwQufnahqlm2w1HQPBO4Gop5",
                    "type": "fanbox.post",
                    "postInfo": {
                        "id": "3964681",
                        "title": "【先行公開】バニーガール円香（差分有）",
                        "feeRequired": "500",
                        "hasAdultContent": true,
                        "creatorId": "singlecask",
                        "user": {
                            "userId": "31699812",
                            "name": "ぷりけつコーギー",
                            "iconUrl": "https://pixiv.pximg.net/c/160x160_90_a2_g5/fanbox/public/images/user/31699812/icon/yiMj9sZ7ZNqmAisF5gQhbxnU.jpeg"
                        },
                        "coverImageUrl": "https://pixiv.pximg.net/c/1200x630_90_a2_g5/fanbox/public/images/post/3964681/cover/ggUQmQO1rcNnsHDVfMgk08I7.jpeg",
                        "excerpt": "グラビア撮影中の円香さん\n何やらご機嫌ナナメのご様子\nどうやら胸のサイズが合ってないようです\nなので別のバニースーツに着替えてもらいましょう\nこれもサイズが合っていないみたいですね～\nと言うことでもう脱いで貰っちゃいましょう\nかわいいおっぱいですね～",
                        "publishedDatetime": "2022-06-10T19:00:00+09:00"
                    }
                }
            }
        },
        "tags": [
            "アイドルマスターシャイニーカラーズ",
            "樋口円香"
        ],
        "excerpt": "ちょっとだけ高画質です\n有料プランはこちら↓",
        "isLiked": false,
        "likeCount": 9,
        "commentCount": 0,
        "restrictedFor": null,
        "isRestricted": false,
        "user": {
            "userId": "31699812",
            "name": "ぷりけつコーギー",
            "iconUrl": "https://pixiv.pximg.net/c/160x160_90_a2_g5/fanbox/public/images/user/31699812/icon/yiMj9sZ7ZNqmAisF5gQhbxnU.jpeg"
        },
        "creatorId": "singlecask",
        "hasAdultContent": true,
        "commentList": {
            "items": [],
            "nextUrl": null
        },
        "nextPost": null,
        "prevPost": {
            "id": "4007886",
            "title": "【支援者様限定】バニー円香【R18】",
            "publishedDatetime": "2022-06-20T19:00:00+09:00"
        },
        "imageForShare": "https://pixiv.pximg.net/c/1200x630_90_a2_g5/fanbox/public/images/post/4012918/cover/u81tmvUMpM0Za5Jk6pkyjSM0.jpeg"
    }
}
```

Perhaps it's a new addition?

**Comment by asdf1324** [comment]
On a sidenote, the downloader gets stuck indefinitely without any indications if there is an error during the parsing:
![image](https://user-images.githubusercontent.com/19871820/175561584-c1f18242-2525-487d-b77f-41fd612b496a.png)

This is what I get if I attempt to download that post, and it does not tell you that an error has occured. Perhaps it would be less confusing if we could display an error message when the extension fails for some reason.

**Comment by xuejianxianzun** [comment]
I fixed this bug in version 1.9.5.
It will take a while to review in the extension store.
https://github.com/xuejianxianzun/PixivFanboxDownloader/releases/tag/v1.9.5

**Comment by asdf1324** [comment]
Thanks - You're doing great, keep up the great work!
